dartagnan/src/main/java/com/dat3m/dartagnan/program/event/visitors/EventVisitor.java

```diff
@@ -34,7 +34,8 @@ public interface EventVisitor<T> {
     default T visitLocal(Local e) { return visitEvent(e); }
     default T visitSkip(Skip e) { return visitEvent(e); }
 
-    default T visitFence(Fence e) { return visitEvent(e); }
+    default T visitTaggedEvent(TaggedEvent e) { return visitEvent(e); }
+    default T visitFence(Fence e) { return visitTaggedEvent(e); }
```

Comment:
I don't get what you mean. `visitFence` is gone because I removed `Fence` (push pending). But it seems you also want to remove `visitTaggedEvent` ...

Reply:
We can keep `visitTaggedEvent` but I'm not sure if that is really a good idea, because the whole point is that you never need to know the class but only its tags (i.e., any check like `instanceof TaggedEvent` or a cast to `TaggedEvent` is HIGHLY suspicious).
But I think it is fine to keep `visitTaggedEvent` because it does have one parameter that is not just a tag, namely its name. That might be enough to justify the existence of dedicated visitor methods for it.